includes/Export.php

```diff
@@ -420,43 +428,39 @@ protected function dumpFrom( $cond = '' ) {
 	 * let's just send a set of small batched queries and yield each row returned. outputPageStream method
 	 * uses foreach to consume the passed $result object.
 	 *
+	 * @see SUS-1395
+	 *
 	 * @param $tables
 	 * @param $vars
 	 * @param $cond
 	 * @param $fname
 	 * @param $opts
 	 * @param $join
-	 * @return ResultWrapper|Generator
+	 * @return Generator
 	 */
 	private function selectRows($tables, $vars, $cond, $fname, $opts, $join) {
-		// we're not doing a full dump, let's just make a single query
-		if ( !empty( $cond ) ) {
-			return $this->db->select( $tables, $vars, $cond, $fname, $opts, $join );
-		}
-		else {
-			// as said above, we do want to make a set of small, buffered queries
-			$this->db->bufferResults( true );
-
-			$page_max = (int) $this->db->selectField( 'page','MAX(page_id)', '', $fname );
+		// as said above, we do want to make a set of small, buffered queries
+		$this->db->bufferResults( true );
 
-			// fetch 500 pages in each batch
-			$offset_start = 1;
-			$batch_size = 500; # this returns up to ~100k rows on yugioh
+		$page_max = (int) $this->db->selectField( 'page','MAX(page_id)', '', $fname );
 
-			while( $offset_start <= $page_max ) {
-				$cond = [
-					sprintf( 'page_id BETWEEN %d AND %d', $offset_start, $offset_start + $batch_size - 1 )
-				];
+		// fetch 500 pages in each batch
+		$offset_start = 1;
+		$batch_size = 500; # this returns up to ~100k rows on yugioh
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)